src/argilla/server/schemas/v1/datasets.py

```diff
@@ -51,3 +54,34 @@ class Annotation(BaseModel):
 
     class Config:
         orm_mode = True
+
+
+class RatingAnnotationSettingsOptionCreate(BaseModel):
+    value: int
+
+
+class RatingAnnotationSettingsCreate(BaseModel):
+    options: conlist(
+        item_type=RatingAnnotationSettingsOptionCreate,
+        min_items=RATING_OPTIONS_MIN_ITEMS,
+        max_items=RATING_OPTIONS_MAX_ITEMS,
+    )
+
+
+class AnnotationCreate(BaseModel):
+    name: str
+    title: str
+    type: AnnotationType
+    required: Optional[bool]
+    settings: Optional[dict] = {}
```

Comment:
I think by using Pydantic discriminators we should define the annotation type field at the settings level. For me, it's fine your solution. Just one thing. If settings are defined as `Optional` that means that we can accept `None` values, but the defined validator will always try to create a new settings instance.
```suggestion
    settings: Union[dict, RatingAnnotationSettingsCreate] = {}
```